index.html

```diff
@@ -648,53 +648,75 @@
 &lt;!-- controller.html --&gt;
 &lt;button id="disconnectBtn" style="display: none;"&gt;Disconnect&lt;/button&gt;
 &lt;button id="stopBtn" style="display: none;"&gt;Stop&lt;/button&gt;
+&lt;button id="reconnectBtn" style="display: none;"&gt;Reconnect&lt;/button&gt;
 &lt;script&gt;
-  var connection;
+  let connection;
 
   // The Disconnect and Stop buttons are visible if there is a connected presentation
-  var disconnectBtn = document.getElementById("disconnectBtn");
-  var stopBtn = document.getElementById("stopBtn");
-  stopBtn.onclick = function () { connection &amp;&amp; connection.terminate(); };
+  let stopBtn = document.querySelector("#stopBtn");
+  let reconnectBtn = document.querySelector("#reconnectBtn");
+  let disconnectBtn = document.querySelector("#disconnectBtn");
+
+  stopBtn.onclick = _ => {
+    connection && connection.terminate();
+  };
 
-  var setConnection = function (theConnection) {
+  disconnectBtn.onclick = _ => {
+    connection && connection.close();
+  };
+
+  function setConnection(newConnection) {
     // Disconnect from existing presentation, if any
-    close();
+    if (connection) {
+      // Termination it not immediate, so run needed steps manually and remove callback
+      connection.onterminate();
+      connection.onterminate = undefined;
+
+      connection.terminate();
```

Comment:
This could terminate the presentation when it's reconnected to, per example 5.3, if the old connection hasn't fired onclose yet.

If we want to guarantee that at most one connection to the presentation is open, then maybe this could be:

```
if (connection && connection != newConnection && connection.state != 'closed') {
  connection.onclosed = undefined;
  connection.close();
}
```